test.disabled/ejabberd_tests/tests/mod_http_notification_SUITE.erl

```diff
@@ -119,15 +119,15 @@ proper_http_message_encode_decode(Config) ->
            after 5000 ->
             ct:fail(http_request_timeout)
            end,
-    ct:log("Encoded :: ~p",[Body]),
-    Decoded = escalus_ejabberd:rpc(cow_qs, urldecode, [Body]),
-    ExpectedMessage = <<"author=",Sender/binary,
-        "&server=",Server/binary,
-        "&receiver=",Receiver/binary,
-        "&message=",Message/binary>>,
-    ct:log("Decoded :: ~p",[Decoded]),
-    ct:log("Expected :: ~p",[ExpectedMessage]),
-    ?assert_equal(ExpectedMessage, Decoded),
+    ExtractedAndDecoded = escalus_ejabberd:rpc(cow_qs, parse_qs,[Body]),
+    ExpectedSender = {<<"author">>,<<Sender/binary>>},
+    ExpectedServer = {<<"server">>,<<Server/binary>>},
+    ExpectedReceiver = {<<"receiver">>,<<Receiver/binary>>},
+    ExpectedMessage = {<<"message">>,<<Message/binary>>},
+    ?assertEqual(lists:member(ExpectedSender, ExtractedAndDecoded), true),
```

Comment:
Actually it may be simpler to just compare lists but sorted. The current code allows `ExtractedAndDecoded` to have more (unknown) fields then expected. Although it could be not relevant, I'm not sure.

Reply:
If we allow to have unknown fields, I'd go with something like

```erlang
ExpectedSet = ordsets:from_list([ExpectedSender, ExpectedServer, ExpectedReceiver, ExpectedMesssage]),
?assert(ordsets:is_subset(ExpectedSet, ordsets:from_list(ExtractedAndDecoded)
```

Reply:
Corollary: `?assert(...)` should be preferred to `?assertEqual(..., true)`

Reply:
Side note: usually in equal assertions, the first argument is the expected value (for correct error information).

Reply:
Or less efficient but simple `[] = [ExpectedSender, ExpectedServer, ExpectedReceiver, ExpectedMessage] -- ExtractedAndDecoded]` :)